maven-artifact/src/test/java/org/apache/maven/artifact/versioning/ComparableVersionTest.java

```diff
@@ -162,21 +173,39 @@ void testVersionsEqual() {
         checkVersionsEqual("1A", "1a");
         checkVersionsEqual("1B", "1b");
         checkVersionsEqual("1M", "1m");
-        checkVersionsEqual("1Ga", "1");
-        checkVersionsEqual("1GA", "1");
-        checkVersionsEqual("1RELEASE", "1");
-        checkVersionsEqual("1release", "1");
-        checkVersionsEqual("1RELeaSE", "1");
-        checkVersionsEqual("1Final", "1");
-        checkVersionsEqual("1FinaL", "1");
-        checkVersionsEqual("1FINAL", "1");
         checkVersionsEqual("1Cr", "1Rc");
         checkVersionsEqual("1cR", "1rC");
         checkVersionsEqual("1m3", "1Milestone3");
         checkVersionsEqual("1m3", "1MileStone3");
         checkVersionsEqual("1m3", "1MILESTONE3");
     }
 
+    @Test
+    void testVersionsEqualOrderAndObjectInequality() {
+        checkVersionsEqualOrder("1ga", "1");
+        checkVersionObjectInequality("1ga", "1");
```

Comment:
Is inequality part of the API for ComparableVersion or is it an implementation detail? If the former it should be added to the class JavaDoc. If the latter, it shouldn't be tested.

Reply:
It was a special behavior before the change, but now that it's not a special behavior, I've removed the test case.